add user data to survey URL (iss. #1216)

lsloan opened the pull request:
Resolves #1216.

Collaboration with @ssciolla and @jonespm.

Adds parameters to the survey URL that will be stored with the survey data.

lsloan committed:
#1216 - add user data to survey URL

Early attempt.  Doesn't contain all the desired information, yet.

jonespm commented:
I updated to fix a conflict here

https://github.com/tl-its-umich-edu/my-learning-analytics/compare/master...jonespm:1216-survey-user-info

It looks like it's working and passing the correct values to the iframe. You may want to also use `user.username `if `user.LTIlaunchID` is not set as we are still supporting SAML as well as local development.

lsloan committed:
#1216 - merge upstream master to get #1215 changes

Merge branch 'master' of https://github.com/tl-its-umich-edu/my-learning-analytics into 1216-survey-user-info

Conflicts:
	assets/src/containers/DashboardAppBar.js

lsloan committed:
#1216 - add username

Per comment by @jonespm, adding user name for those times MyLA wasn't launched via LTI and the LTI launch ID isn't available.

lsloan committed:
#1216 - fix formatting

ssciolla committed:
Add draft of full implementation; ensure SurveyModal conditionally disappears

lsloan committed:
Merge pull request #2 from ssciolla/issue-1216-survey-params

Add draft of full implementation; ensure SurveyModal conditionally di…

ssciolla commented on assets/src/components/SurveyModal.js:
```diff
@@ -50,8 +51,25 @@ const useStyles = makeStyles((theme) => ({
   const [open, setOpen] = useState(false)
 
+  const lastTwoSteps = location.pathname.split('/').slice(-2)
+  const viewName = lastTwoSteps.length < 2
+    ? ''
```
This was just a fallback (maybe it should be `unknown`?). I don't love this approach since it could easily break if you changed the application structure, but that seems unlikely to happen for now.

I guess the alternative would be passing hardcoded values as props wherever `DashboardAppBar` is used. That could be more flexible, but would require updating a handful more files.

ssciolla replied:
Alternatively, we could just pass the whole `pathname`.

lsloan replied:
yeah, something like `location.pathname || 'unknown'`.

lsloan replied:
That's not important, though.  It'd be nice to have, but it can be changed in a future version if it's decided that's a better approach.

lsloan commented:
@zqian, @pushyamig, & @chrisrrowland: because @jonespm and @ssciolla collaborated with me on this issue, I think you three are the only developers that should review and approve this PR.  I've added your names as reviewers.  If you're unable to review this, please remove your name from the list of reviewers and tell the remaining viewers that you're leaving the task with them.

Thanks in advance!

lsloan commented:
@ssciolla suggested I review it so we can make progress.

GitHub doesn't think I should review it.  It won't let me select my name as a reviewer, but I'll start a review anyway.

lsloan commented on assets/src/components/SurveyModal.js:
```diff
@@ -50,8 +51,25 @@ const useStyles = makeStyles((theme) => ({
+    courseID: props.courseID,
+    view: viewName
+  }
+  const searchParams = new URLSearchParams()
+  Object.entries(params).map(([key, value]) => searchParams.append(key, value))
```
Very nice.  I'd hoped there would be something like this available.  I didn't really like my initial approach, concatenating a string myself.

lsloan reviewed (commented):
Thanks for the major assist, @ssciolla!
